.github/copilot-instructions.md

```diff
@@ -0,0 +1,197 @@
+  - **Requires Chrome/Selenium setup**
+
+### Linting and Code Quality
+- Ruby linting: `bundle exec rubocop --except Metrics`
+  - Auto-fix: `bundle exec rubocop --except Metrics -a`
+- JavaScript linting: `npx eslint app/javascript`
+- HAML linting: `bundle exec haml-lint`
+- **ALWAYS run linting before committing or CI will fail**
+
+### Running the Development Server
+Use the Procfile.dev to orchestrate all services:
+
+#### Mac-specific
+```bash
+hivemind -p 3020 Procfile.dev
+```
+
+#### Unix-specific  
+```bash
+overmind start --port 3020 --procfile Procfile.dev
+```
+
+#### Manual startup (all platforms)
+```bash
+# Terminal 1: Rails server
+bundle exec rails server -p 3020
+
+# Terminal 2: AnyCable
+bundle exec anycable
+
+# Terminal 3: Sidekiq
+bundle exec sidekiq
+
+# Terminal 4: WebSocket server
+anycable-go --host='local.exercism.io' --rpc_host='local.exercism.io:50051' --port=3334
+
+# Terminal 5: CSS build (watch mode)
+yarn build:css --watch
+
+# Terminal 6: JavaScript build (watch mode)
+yarn build --watch
+```
+
+## Validation and Testing
+
+### Manual Validation Requirements
+After making changes, ALWAYS test core functionality:
+1. Browse to http://localhost:3020
+2. Test user registration/login flow
+3. Navigate key sections (tracks, exercises, community)
+4. Test JavaScript interactions (code editor, modals)
+5. Verify CSS styling renders correctly
+
+### Pre-commit Validation
+ALWAYS run before committing:
+```bash
+bundle exec rubocop --except Metrics
+yarn test
+bundle exec rails test:zeitwerk
+```
+
+## Critical Warnings and Limitations
+
+### Build Timing - NEVER CANCEL
+- **Bundle install: 15-20 minutes** - Native extensions for grpc, skylight, commonmarker take time
+- **Ruby test suite: 10-15 minutes** - Full suite with database operations
+- **System tests: 15-20 minutes** - Browser automation with Capybara/Selenium
+- **Asset builds: 2-5 minutes each** - CSS and JavaScript compilation
+
+### Authentication Requirements
+- **NPM Token required**: Private package access needs `NPM_TOKEN` environment variable
+- **AWS LocalStack**: Required for S3, DynamoDB, and other AWS service mocking
+- **Database credentials**: Must match config/database.yml settings
+
+### Version Dependencies (CRITICAL)
+- **Ruby 3.4.4**: Exact version required - Gemfile.lock specifies this
+- **Node.js 20+**: Required for esbuild and modern JS features
+- **MySQL 5.7**: Database schema depends on this version
+- **Bundler 2.6.9**: Locked version in Gemfile.lock
+
+### Known Limitations
+- Ruby version managers (rbenv/rvm) recommended for version management
+- Some JS packages require private NPM access (will fail in some environments)  
+- Docker services must be running before Rails server starts
+- Database must be properly configured with utf8mb4 collation
+- Large codebase - initial setup can take 30+ minutes total
+
+## Important Directories
+
+### Application Structure
+- `app/controllers/` - Rails controllers
```

Comment:
Add a section explaining the API, SPI and normal routes. 

Explain how API delegates to Mandate commands
Explain how SPI is a private API used internally and secured at the AWS level, allowing things like Lambdas to post data back arbitarily.